### DISCUSSION TITLE
How to build .deb package

### DISCUSSION THREAD

**Discussion by ar0311** [ravendb/ravendb#14239] [Q&A]
Hi

Just looking for some direction on building deb package from custom build

Cheers

**Comment by gregolsky** [comment]
Hello,

We do not have an official guide there. Here are the scripts we use for building DEBs though:

https://github.com/ravendb/ravendb/tree/v5.3/scripts/linux/pkg/deb

Prerequisites is Docker for the most part and some qemu assets for building for other architectures that x64 if you need that.

The script builds the package based on an existing and already published tarball. Here's where it pulls the RavenDB tarball, so in order to make a package from a custom build, you would need to supply it there in the pipeline:
https://github.com/ravendb/ravendb/blob/f773b7da6e8afccad88549ee528fd1165878baa1/scripts/linux/pkg/deb/assets/build.sh#L25

**Reply by ar0311** [reply]
OK, would you be able to provide guidance on creating a custom docker image for testing?

**Reply by gregolsky** [reply]
To build a custom docker image you can use the following PS script: 
https://github.com/ravendb/ravendb/blob/v6.0/docker/build-ubuntu.ps1 

and point its ArtifactsDir to directory of your choice plus make sure it's able to match the package file name.
